var/spack/repos/builtin/packages/h5bench/package.py

```diff
@@ -58,3 +58,32 @@ def cmake_args(self):
         ]
 
         return args
+
+    def check(self):
+        if self.run_tests:
+            with working_dir(self.build_directory):
+                make("test")
```

Comment:
You can remove this.  Spack will automatically run `make check` and or `make test` when present for you since you're inheriting from `CMakePackage`.  (See https://spack.readthedocs.io/en/latest/packaging_guide.html#adding-installation-phase-tests.)

Reply:
If you wanted to something else, that section of the docs has examples of post-build and post-install tests.  

You probably don't want to override the ~reserved `check` method. 

You would access the build directory with to `self.builder.build_directory` now that multiple build systems are supported.